include/vctr/Containers/Span.h

```diff
@@ -106,6 +106,9 @@ class Span : public VctrBase<ElementType, std::span<ElementType, extent>, extent
     //==============================================================================
     // Operators
     //==============================================================================
+    // Deleting the default copy operator forces the usage of the intended operator below
+    Span& operator= (const Span&) = delete;
```

Comment:
This must be what fixes the problems when assigning one span to another

Reply:
Indeed it fixes the following tests (section 1 and 3 did fail). Can you please add them or equivalent tests?

```
TEST_CASE ("Span assignments")
{
    vctr::Array<int, 1> a (0);
    vctr::Array<int, 1> b (1);

    SECTION ("Attempt 1")
    {
        a.subSpan (0) = b.subSpan (0);
        REQUIRE (a[0] == b[0]); // fails locally
    }

    SECTION ("Attempt 2")
    {
        auto aSpan = vctr::Span (a.data(), a.size());
        auto bSpan = vctr::Span (b.data(), b.size());
        aSpan = bSpan;

        REQUIRE (a[0] == b[0]); // passes locally
    }

    SECTION ("Attempt 3")
    {
        auto makeSpan = [] (int* ptr, size_t size) { return vctr::Span (ptr, size); };

        makeSpan (a.data(), a.size()) = makeSpan (b.data(), b.size());

        REQUIRE (a[0] == b[0]); // fails locally
    }

    SECTION ("Attempt 4")
    {
        auto makeSpan = [] (int* ptr, size_t size) { return vctr::Span (ptr, size); };

        makeSpan (a.data(), a.size())[0] = makeSpan (b.data(), b.size())[0];

        REQUIRE (a[0] == b[0]); // passes locally
    }
}
```

Reply:
It is. Surprisingly simple 🙈

Reply:
I added an issue for that, since I'd like to add a few more assignment tests https://github.com/sonible/VCTR/issues/90